tools/lsp/preview.rs

```diff
@@ -561,48 +561,47 @@ fn show_preview_for(name: slint::SharedString, url: slint::SharedString) {
 }
 
 // triggered from the UI, running in UI thread
-fn can_drop_component(
-    component_type: slint::SharedString,
-    x: f32,
-    y: f32,
-    on_drop_area: bool,
-) -> bool {
+fn can_drop_component(component_index: i32, x: f32, y: f32, on_drop_area: bool) -> bool {
     if !on_drop_area {
         set_drop_mark(&None);
         return false;
     }
 
     let position = LogicalPoint::new(x, y);
-    let component_type = component_type.to_string();
+    let component_type = PREVIEW_STATE.with(|preview_state| {
+        let preview_state = preview_state.borrow();
+
+        preview_state
+            .known_components
+            .get(component_index as usize)
+            .map(|ci| ci.name.clone())
+            .unwrap_or(String::new())
```

Comment:
Wouldn't it make sense to return an Option instead and return false for `can_drop_component`, instead of going all the way into can_drop_at?